src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -861,7 +861,16 @@ def _revoked_certificates(self):
             for i in range(num):
                 r = self._backend._lib.sk_X509_REVOKED_value(revoked, i)
                 self._backend.openssl_assert(r != self._backend._ffi.NULL)
-                revoked_list.append(_RevokedCertificate(self._backend, r))
+                revoked_certificate = _RevokedCertificate(self._backend, r)
+                # The X509_REVOKED_value is a X509_REVOKED * that has
+                # no reference counting. This means when X509_CRL_free is
+                # called then the CRL and all X509_REVOKED * are freed. Since
+                # you can retain a reference to a single revoked certificate
+                # and let the CRL fall out of scope, we need to retain a
+                # private reference to the CRL inside the RevokedCertificate
+                # object to prevent the gc from being called inappropriately.
+                revoked_certificate._crl = self
```

Comment:
Please put this in the constructor, rather than hanging it off in random places.